homeassistant/components/coolmaster/climate.py

```diff
@@ -151,6 +172,15 @@ async def async_set_fan_mode(self, fan_mode: str) -> None:
         self._unit = await self._unit.set_fan_speed(fan_mode)
         self.async_write_ha_state()
 
+    async def async_set_swing_mode(self, swing_mode: str) -> None:
+        """Set new swing mode."""
+        _LOGGER.debug("Setting swing mode of %s to %s", self.unique_id, swing_mode)
+        try:
+            self._unit = await self._unit.set_swing(swing_mode)
+        except Exception as error:
```

Comment:
Limit this to specific types that you expect to throw here such as connection errors relevant to the specific library etc

Reply:
The `Exception` catch is since it can be raised by the call [here](https://github.com/OnFreund/pycoolmasternet-async/blob/master/pycoolmasternet_async/coolmasternet.py#L41). This is a limitation of the underline library which uses the base class. Do you see a better approach here?

Reply:
If that's an expected case that can happen (e.g. transient communication failure, something the user can do to put it into that state, etc) then we need to fix the library to throw a more specific exception. My impression is this library is changeable give you've been committing changes to it. If this is not expected to happen (e.g. corner cases that shouldn't happen) then we can catch a more narrow exception that only covers the expected cases and the exception handling can be cleaned up without blocking this.

Reply:
Done.

Reply:
How does `ValueError` happen? That also seems like it would be catching a coding bug...

(I noticed that there isn't really any exception handling for the other calls either -- but i would have expected asyncio related errors here talking to the device as the only type of exception to catch. Since none of the other client rpcs are handling those, it seems like something that can be improved in a followup to move this integration up the quality scale)

Reply:
Yes, this is a real one. It can happen when the A/C unit doesn't support this mode (and there is no way to query the valid modes up front). The code is [here](https://github.com/OnFreund/pycoolmasternet-async/blob/master/pycoolmasternet_async/coolmasternet.py#L194).

Reply:
I think the library should be changed to raise a more specific exception. In my (very weak) defense, this was probably there since before I forked it.
@amitfin if you create a PR against the library I'll do my best to merge and release quickly.

Reply:
OK. As follow ups, I recommend overhauling exception handling in this library to stop throwing `Exception` and consider improving this one too.